tutorial/practicals/nemo/2_nemo_profiling/README.md

```diff
@@ -0,0 +1,183 @@
+   profiling automatically inserted around each routine:
+
+       $ make tra_adv.exe
+
+   When examining the generated Fortran code (in `psy.f90`), you
+   should see that PSyclone has added `USE profile_psy_data_mod, ONLY:
+   profile_PSyDataType` as well as calls to
+   `profile_psy_data0%PreStart` and `profile_psy_data0%PostEnd`.
+   Since the code now depends upon the PSyData API, the location of a
+   suitable wrapper library must be provided when compiling the
+   mini-app.  The supplied Makefile will build the 'simple_timing'
+   implementation of this library and link our mini-app against it.
+
+   At this point, the compiled application can be run:
+
+       $ ./tra_adv.exe
+
+   but no timing information is output!
+
+2. No timing information is output because the timing library is not
+   being shutdown. This is because PSyclone does not have any
+   knowledge of the overall structure of an application and therefore
+   it is up to the user to insert startup/shutdown calls in the
+   appropriate locations (e.g. before/after MPI is
+   initialised/finalised).  The 'simple_timing' library only requires
+   a shutdown call so you will need to edit the driver program
+   (`runner.f90`), add a `USE profile_psy_data_mod, only:
+   profile_psydatashutdown` at the beginning and then add a call,
+   `CALL profile_psydatashutdown()` as the final statement before the
+   `END PROGRAM`. (This is the reason that the mini-app has been
+   restructured slightly for this tutorial.)
+
+   Rebuild the application (`make tra_adv.exe`) and run it. You should now
+   see timing information printed to the terminal, e.g.:
+
+       ===========================================
+       module::region   count	sum	   min		average		max
+       tra_adv::r0        1   0.718750000    0.718750000   0.718750000    0.718750000
+       ===========================================
+
+   Timings are only reported for a single region because our mini-app consists
+   of a single subroutine.
+
+   If you examine the Makefile, you will see that PSyclone has been run with
+   the `--profile invokes` option and it is this that causes the subroutine
+   to be instrumented for profiling.
+
+3. Edit the Makefile so that the line invoking PSyclone uses the
+   `--profile kernels` option instead. Rebuild and run the mini-app. You
+   should now see timing for four regions that have been identified as
+   kernels by PSyclone:
+   
+       ===========================================
+       module::region   count	sum		min		average         max
+       tra_adv::r0        1    3.12500000E-02   3.12500000E-02 	3.12500000E-02 	3.12500000E-02
+       tra_adv::r1        1    0.00000000       0.00000000     	0.00000000     	0.00000000
+       tra_adv::r2        1    0.00000000       0.00000000     	0.00000000     	0.00000000    
+       tra_adv::r3        1    0.343750000      0.343750000     0.343750000     0.343750000
```

Comment:
Oh yes, so do I. This is a bug because we're identifying a loop containing a CodeBlock as an InlinedKern. I thought I had at least an issue open for this but it appears I don't. I'll create one.

Reply:
#983 addresses this.

Reply:
OK. Probably worth double checking when #983 gets merged to master.